src/kbmod/search/kernels.cu

```diff
@@ -86,9 +86,21 @@ __device__ float readEncodedPixel(void *imageVect, int index, int numBytes, cons
 __global__ void searchFilterImages(int imageCount, int width, int height, void *psiVect, void *phiVect,
                                    perImageData image_data, searchParameters params, int trajectoryCount,
                                    trajectory *trajectories, trajectory *results) {
-    // Get origin pixel for the trajectories.
-    const unsigned short x = blockIdx.x * THREAD_DIM_X + threadIdx.x;
-    const unsigned short y = blockIdx.y * THREAD_DIM_Y + threadIdx.y;
+    // Get the x and y coordinates within the search space.
+    const int x_i = blockIdx.x * THREAD_DIM_X + threadIdx.x;
+    const int y_i = blockIdx.y * THREAD_DIM_Y + threadIdx.y;
+
+    // Check that the x and y coordinates are consistent with the search space.
+    const int search_width = params.x_start_max - params.x_start_min;
+    const int search_height = params.y_start_max - params.y_start_min;
```

Comment:
Are the sizes of search area something we might want at a higher level of the code too? Should we move this above?

Reply:
The sizes are used in other parts of the code. Since it is just a pair of subtractions, I don't think it is a huge overhead to recompute them from the params in a few different functions.